client/app/visualizations/chart/plotly.js

```diff
@@ -499,6 +499,9 @@ const CustomPlotlyChart = (clientConfig) => {
         return;
       }
       const refresh = () => {
+        // Clear existing data
+        element[0].children[0].calcdata = [];
+        element[0].children[0].data = [];
```

Comment:
Please see https://github.com/getredash/redash/pull/1873/commits/1a8078ab0345108ef91ca5a17835b3c6816f15ba .

This less hackish way fixes the problem too.